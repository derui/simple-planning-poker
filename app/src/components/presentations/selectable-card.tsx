import classnames from "classnames";
import { BaseProps, generateTestId } from "../base";

interface Props extends BaseProps {
  display: string;
  selected: boolean;
  onSelect: () => void;
}

const styles = {
  root: (selected: boolean) =>
    classnames(
      "flex",
      "flex-col",
      "h-20",
      "w-14",
      "rounded",
      "border",
      "border-primary-400",
      "text-center",
      "justify-center",
      "m-3",
      "transition-transform",
      "first-of-type:ml-0",
      "last-of-type:mr-0",

      "hover:-translate-y-2",
      {
        "bg-white": !selected,
        "text-primary-400": !selected,
      },
      {
        "bg-primary-400": selected,
        "text-secondary1-200": selected,
        "-translate-y-2": selected,
      }
    ),
};

// eslint-disable-next-line func-style
export function SelectableCard(props: Props) {
  const gen = generateTestId(props.testid);

  return (
    <div
      className={styles.root(props.selected)}
      onClick={props.onSelect}
      data-testid={gen("root")}
      data-selected={props.selected}
    >
      {props.display}
    </div>
  );
}
