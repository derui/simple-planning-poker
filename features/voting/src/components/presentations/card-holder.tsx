import { PropsWithChildren } from "react";
import * as styles from "./card-holder.css.js";

export interface Props {
  /**
   * type of holder
   */
  type: "player" | "inspector";
}

export const CardHolder = function CardHolder({ type, children }: PropsWithChildren<Props>): JSX.Element {
  return (
    <div className={styles.root} role="tablist">
      {type == "player" ? children : <p className={styles.inspector}>Inspector mode</p>}
    </div>
  );
};
