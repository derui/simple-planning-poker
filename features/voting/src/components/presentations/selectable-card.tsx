import { PropsWithChildren } from "react";
import * as styles from "./selectable-card.css.js";

interface Props {
  selected?: boolean;
  onSelect?: () => void;
}

// eslint-disable-next-line func-style
export function SelectableCard({ selected = false, children, onSelect }: PropsWithChildren<Props>) {
  return (
    <div
      className={selected ? styles.selected : styles.notSelected}
      onClick={onSelect}
      data-selected={selected}
      role="tab"
      aria-selected={selected}
    >
      {children}
    </div>
  );
}
