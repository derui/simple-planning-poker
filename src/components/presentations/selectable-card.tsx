import classnames from "classnames";

interface Props {
  storyPoint: string;
  selected: boolean;
  onSelect: () => void;
}

const styles = {
  root: (selected: boolean) =>
    classnames(
      "flex",
      "flex-col",
      "h-12",
      "w-6",
      "rounded",
      "border",
      "border-primary-400",
      "bg-white",
      "text-primary-400",
      "text-center",
      "justify-center",
      "m-3",
      "transition-transform",
      "first-of-type:ml-0",
      "last-of-type:mr-0",

      "hover:[transform:transformY(calc(-1 * 0.5rem))]",
      {
        "bg-primary-400": selected,
        "text-secondary1-200": selected,
        "[transform:transformY(calc(-1 * 0.5rem))]": selected,
      }
    ),
};

// eslint-disable-next-line func-style
export function SelectableCard(props: Props) {
  return (
    <div className={styles.root(props.selected)} onClick={props.onSelect} data-selected={props.selected}>
      {props.storyPoint}
    </div>
  );
}
