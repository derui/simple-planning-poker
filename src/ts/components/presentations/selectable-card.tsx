import { Component } from "solid-js";

interface Props {
  display: string;
  selected: boolean;
  onClick: () => void;
}

export const SelectableCard: Component<Props> = (props) => {
  const className = {
    "app__game__selectable-card": true,
    "app__game__selectable-card--selected": props.selected,
  };

  return (
    <div classList={className} onClick={props.onClick}>
      <span class="app__game__selectable-card__storypoint">{props.display}</span>
    </div>
  );
};
