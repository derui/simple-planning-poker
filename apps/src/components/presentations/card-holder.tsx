import classNames from "classnames";
import { BaseProps, generateTestId } from "../base";
import { SelectableCard } from "./selectable-card";

interface Props extends BaseProps {
  displays: string[];
  selectedIndex: number | null;
  onSelect: (index: number) => void;
}

const styles = {
  root: classNames("flex", "flex-auto", "justify-center", "px-3", "py-2", "h-36", "items-end"),
};

const createCard = (
  display: string,
  index: number,
  selectedIndex: number | null,
  onClickCard: (index: number) => void,
  testid: string
) => {
  return (
    <SelectableCard
      testid={testid}
      key={index}
      display={display}
      selected={index === selectedIndex}
      onSelect={() => onClickCard(index)}
    />
  );
};

// eslint-disable-next-line func-style
export function CardHolder(props: Props) {
  const gen = generateTestId(props.testid);

  return (
    <div className={styles.root} data-testid={gen("root")}>
      {props.displays.map((display, index) =>
        createCard(display, index, props.selectedIndex, props.onSelect, gen("card"))
      )}
    </div>
  );
}
