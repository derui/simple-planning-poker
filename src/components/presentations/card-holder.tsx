import React from "react";
import { SelectableCard } from "./selectable-card";

interface Props {
  displays: string[];
  selectedIndex: number | null;
  onClickCard: (index: number) => void;
}

const createCard = (
  display: string,
  index: number,
  selectedIndex: number | null,
  onClickCard: (index: number) => void
) => {
  return (
    <SelectableCard
      key={index}
      storyPoint={display}
      selected={index === selectedIndex}
      onSelected={() => onClickCard(index)}
    />
  );
};

export const CardHolderComponent: React.FunctionComponent<Props> = (props) => {
  return (
    <div className="app__game__card-holder">
      {props.displays.map((display, index) => createCard(display, index, props.selectedIndex, props.onClickCard))}
    </div>
  );
};
