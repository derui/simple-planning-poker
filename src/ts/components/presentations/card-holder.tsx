import React from "react";
import { SelectableCardComponent } from "./selectable-card";

interface Props {
  displays: string[];
  selectedIndex: number | null;
  onClickCard: (index: number) => void;
}

const createCard = (
  display: string,
  index: number,
  selectedIndex: number | null,
  onClickCard: (index: number) => void
) => {
  return (
    <SelectableCardComponent
      key={index}
      display={display}
      selected={index === selectedIndex}
      onClick={() => onClickCard(index)}
    />
  );
};

export const CardHolderComponent: React.FunctionComponent<Props> = (props) => {
  return (
    <div className="app__game__card-holder">
      {props.displays.map((display, index) => createCard(display, index, props.selectedIndex, props.onClickCard))}
    </div>
  );
};
