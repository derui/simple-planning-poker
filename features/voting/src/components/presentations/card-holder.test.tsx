import { test, afterEach, expect } from "vitest";
import { render, cleanup } from "@testing-library/react";
import { CardHolder } from "./card-holder.js";
import { SelectableCard } from "./selectable-card.js";

afterEach(cleanup);

test("should be able to render", async () => {
  const ret = render(
    <CardHolder type="player">
      <SelectableCard>3</SelectableCard>
    </CardHolder>
  );

  expect(ret.container).toMatchSnapshot();
});

test("should be able to render as inspector", async () => {
  const ret = render(
    <CardHolder type="inspector">
      <SelectableCard>3</SelectableCard>
    </CardHolder>
  );

  expect(ret.container).toMatchSnapshot();
});
