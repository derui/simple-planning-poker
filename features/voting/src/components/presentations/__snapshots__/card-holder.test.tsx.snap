// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should be able to render 1`] = `
<div>
  <div
    class="card-holder_root__j7wifn0"
    role="tablist"
  >
    <div
      aria-selected="false"
      class="selectable-card_notSelected__t2s9ip1 selectable-card_rootBase__t2s9ip0 style_transition_all__59wy0241"
      data-selected="false"
      role="tab"
    >
      3
    </div>
  </div>
</div>
`;

exports[`should be able to render as inspector 1`] = `
<div>
  <div
    class="card-holder_root__j7wifn0"
    role="tablist"
  >
    <p
      class="card-holder_inspector__j7wifn1"
    >
      Inspector mode
    </p>
  </div>
</div>
`;
